ipatests/test_integration/test_installation.py

```diff
@@ -479,3 +481,51 @@ def test_reserved_ip_as_forwarder(self):
         exp_str = ("Invalid IP Address 0.0.0.0: cannot use IANA reserved "
                    "IP address 0.0.0.0")
         assert exp_str in cmd.stdout_text
+
+
+class TestKRAinstallAfterCertRenew(IntegrationTest):
+    """ Test KRA installtion after ca agent cert renewal
+
+    KRA installtion was failing after ca-agent cert gets renewed.
+    This test check if the syptoms no longer exists.
+
+    related ticket: https://pagure.io/freeipa/issue/7288
+    """
+
+    def test_KRA_install_after_cert_renew(self):
+
+        tasks.install_master(self.master)
+
+        # get ca-agent cert contents
+        p12_content = self.master.get_file_contents(paths.DOGTAG_ADMIN_P12)
+        admin_pass = self.master.config.admin_password
+
+        # convert cert to pem format
+        p12 = crypto.load_pkcs12(p12_content, admin_pass)
+        cert = crypto.dump_certificate(crypto.FILETYPE_PEM,
+                                       p12.get_certificate())
+
+        # get expiry date of agent cert
+        x509 = crypto.load_certificate(crypto.FILETYPE_PEM, cert)
+        cert_expiry = x509.get_notAfter()
+
+        # remove the 'Z' from the date string
+        cert_expiry = cert_expiry[:-1].decode('utf-8')
+        py_date_obj = datetime.strptime(cert_expiry, "%Y%m%d%H%M%S")
+
+        # move date to grace period so that certs get renewed
+        grace_date = py_date_obj - timedelta(days=20)
+        grace_date = datetime.strftime(grace_date, "%Y-%m-%d %H:%M:%S")
+        self.master.run_command(['date', '-s', grace_date])
+
+        # check if all certs renewed
+        count = 0
+        while (count != 9):
+            cmd = self.master.run_command(['getcert', 'list'])
+            count = cmd.stdout_text.count('status: MONITORING')
+
```

Comment:
I don't understand, why is this while set precisely to count !=9, please?

Reply:
This is to check if all certs got renewed. Cert count is 9, that's why it is checking the status count for 9. If count is 9, it means certs got renewed and we are good go next step.